plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -43,6 +43,9 @@
  */
 public class GoogleLoginService implements IGoogleLoginService {
 
+  private static final String OAUTH_DATA_STORE_PREFERENCE_PATH =
```

Comment:
Yeah, @nbashirbello was also talking a very similar thing (although what she was talking about was to having a single node, no children, and have things like `prefs.put("KEY_EMAIL" + credential.getStoredEmail(), ...)`.

The reason I hesitated a bit now is that, in IntelliJ, we sometimes observe an email is null. I need to look into that to figure out why it's sometimes null.